src/Script/Composer/Composer.php

```diff
@@ -6,11 +6,23 @@
 
 class Composer
 {
-    public static function removeUnusedServices(
-        Event      $event,
-        Filesystem $filesystem = null
-    )
+
+    public static function onlyRemoveUnusedServicesInDev(Event $event, Filesystem $filesystem = null)
     {
+        self::removeUnusedServicesByExtraConfig($event, $filesystem, true);
+    }
+
+    public static function removeUnusedServices(Event $event, Filesystem $filesystem = null)
+    {
+        self::removeUnusedServicesByExtraConfig($event, $filesystem, false);
+    }
+
+    private static function removeUnusedServicesByExtraConfig(Event $event, Filesystem $filesystem = null, $onlyRunInDev = false)
+    {
+        if ($onlyRunInDev && !$event->isDevMode()){
```

Comment:
What happens when a customer is in dev mode and  `isDevMode()` is called from an older version of composer?  I see this was introduced 4 years ago.

Reply:
I noticed that the minimum version of the `composer/composer` package required for this package is `1.10.22`, and I can make sure that `isDevMode ()` exists.
https://github.com/composer/composer/blame/1.10.22/src/Composer/Script/Event.php

Reply:
See this [PR](https://github.com/composer/composer/commit/4b8813269c3d365285512f2386e25cea528ee2c4#diff-ec575f0ac8615348140bec4bbf83eb44b52e1c0b41251f712147927329309d9c)  that on composer `2.5.4` version